### ISSUE TITLE
[facebook] username/posts/ link type detection in extractor Unsupported URL

### ISSUE THREAD

**Issue by someziggyman** [yt-dlp/yt-dlp#3191]
### Checklist

- [X] I'm reporting a site feature request
- [X] I've verified that I'm running yt-dlp version **2022.03.08.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

anywhere

### Example URLs

https://www.facebook.com/gilbert.mennetretmoya/posts/pfbid02ALGfNB6ukGTHbooC4rGAiG3GRjyiTaGGzTVBQtd9fXBGXT5K6imKTwxm8vavVv26l

### Description

Ok ,so, there's the link type with URL scheme Facebook/username/posts that leads to a post with a video in it.

for some reason it's handled by Generic extractor, though I guess Facebook.py should handle this. 
If right clicked on the video, I'm getting this link and it works:
https://www.facebook.com/gilbert.mennetretmoya/videos/643373690059958/

It opens on Facebook watch in browser, while the initial /posts/ link opens as regular FB post page. 

### Verbose log

```shell
yt-dlp -v -F https://www.facebook.com/gilbert.mennetretmoya/posts/pfbid02ALGfNB6ukGTHbooC4rGAiG3GRjyiTaGGzTVBQtd9fXBGXT5K6imKTwxm8vavVv26l
[debug] Command-line config: ['-v', '-F', 'https://www.facebook.com/gilbert.mennetretmoya/posts/pfbid02ALGfNB6ukGTHbooC4rGAiG3GRjyiTaGGzTVBQtd9fXBGXT5K6imKTwxm8vavVv26l']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.03.08.1 [28469edd7] (zip)
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python version 3.9.10 (CPython 64bit) - macOS-11.6.5-arm64-arm-64bit
[debug] exe versions: none
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
[debug] [generic] Extracting URL: https://www.facebook.com/gilbert.mennetretmoya/posts/pfbid02ALGfNB6ukGTHbooC4rGAiG3GRjyiTaGGzTVBQtd9fXBGXT5K6imKTwxm8vavVv26l
[generic] pfbid02ALGfNB6ukGTHbooC4rGAiG3GRjyiTaGGzTVBQtd9fXBGXT5K6imKTwxm8vavVv26l: Requesting header
[redirect] Following redirect to https://www.facebook.com/login/?next=https%3A%2F%2Fwww.facebook.com%2Fgilbert.mennetretmoya%2Fposts%2Fpfbid02ALGfNB6ukGTHbooC4rGAiG3GRjyiTaGGzTVBQtd9fXBGXT5K6imKTwxm8vavVv26l
[debug] [generic] Extracting URL: https://www.facebook.com/login/?next=https%3A%2F%2Fwww.facebook.com%2Fgilbert.mennetretmoya%2Fposts%2Fpfbid02ALGfNB6ukGTHbooC4rGAiG3GRjyiTaGGzTVBQtd9fXBGXT5K6imKTwxm8vavVv26l
[generic] ?next=https://www.facebook.com/gilbert: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] ?next=https://www.facebook.com/gilbert: Downloading webpage
[generic] ?next=https://www.facebook.com/gilbert: Extracting information
[debug] Looking for video embeds
[redirect] Following redirect to https://www.facebook.com/login/?next=https%3A%2F%2Fwww.facebook.com%2Fgilbert.mennetretmoya%2Fposts%2Fpfbid02ALGfNB6ukGTHbooC4rGAiG3GRjyiTaGGzTVBQtd9fXBGXT5K6imKTwxm8vavVv26l&_fb_noscript=1
[debug] [generic] Extracting URL: https://www.facebook.com/login/?next=https%3A%2F%2Fwww.facebook.com%2Fgilbert.mennetretmoya%2Fposts%2Fpfbid02ALGfNB6ukGTHbooC4rGAiG3GRjyiTaGGzTVBQtd9fXBGXT5K6imKTwxm8vavVv26l&_fb_noscript=1
[generic] ?next=https://www.facebook.com/gilbert: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] ?next=https://www.facebook.com/gilbert: Downloading webpage
[generic] ?next=https://www.facebook.com/gilbert: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://www.facebook.com/login/?next=https%3A%2F%2Fwww.facebook.com%2Fgilbert.mennetretmoya%2Fposts%2Fpfbid02ALGfNB6ukGTHbooC4rGAiG3GRjyiTaGGzTVBQtd9fXBGXT5K6imKTwxm8vavVv26l&_fb_noscript=1
Traceback (most recent call last):
  File "/Users/zig/yt-dlp/./yt-dlp/yt_dlp/YoutubeDL.py", line 1398, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/zig/yt-dlp/./yt-dlp/yt_dlp/YoutubeDL.py", line 1468, in __extract_info
    ie_result = ie.extract(url)
  File "/Users/zig/yt-dlp/./yt-dlp/yt_dlp/extractor/common.py", line 638, in extract
    ie_result = self._real_extract(url)
  File "/Users/zig/yt-dlp/./yt-dlp/yt_dlp/extractor/generic.py", line 4017, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.facebook.com/login/?next=https%3A%2F%2Fwww.facebook.com%2Fgilbert.mennetretmoya%2Fposts%2Fpfbid02ALGfNB6ukGTHbooC4rGAiG3GRjyiTaGGzTVBQtd9fXBGXT5K6imKTwxm8vavVv26l&_fb_noscript=1
```

**Comment by pukkandan** [comment]
I can't reproduce. Depending on your IP, you may need login to watch facebook videos. Check if you can open the link in an incognito window. If not, you'll need to pass cookies

```
❯ yt-dlp https://www.facebook.com/gilbert.mennetretmoya/videos/643373690059958/
[facebook] 643373690059958: Downloading webpage
[facebook] 643373690059958: Downloading MPD manifest
[info] 643373690059958: Downloading 1 format(s): 1711595572565770v-1+994051227984108a-1
[download] Destination: 🙃🙃🙃 [643373690059958].f1711595572565770v-1.mp4
[download] 100% of 845.63KiB in 00:01
[download] Destination: 🙃🙃🙃 [643373690059958].f994051227984108a-1.m4a
[download] 100% of 102.41KiB in 00:00
[Merger] Merging formats into "🙃🙃🙃 [643373690059958].mp4"
Deleting original file 🙃🙃🙃 [643373690059958].f1711595572565770v-1.mp4 (pass -k to keep)
Deleting original file 🙃🙃🙃 [643373690059958].f994051227984108a-1.m4a (pass -k to keep)
```

**Comment by pukkandan** [comment]
gotcha, my bad

**Comment by dausruddin** [comment]
Still an issue on version `2022.06.29`

**Comment by codenamedpkt** [comment]
Me too in the latest master branch I have downloaded and passed like this:

`yt-dlp --cookies yourfb.cookies -o your/folder/outputs/%(filename)s.%(ext)s https://www.facebook.com/username/videos_by https://www.facebook.com/ktpoon.is.no.1/photos_albums`

and gets unsupported URL while downloading.

**Comment by bashonly** [comment]
Fixed by cf9af2c7f1fedd881a157b3fbe725e5494b00924
